reader_test.go

```diff
@@ -337,3 +337,40 @@ func BenchmarkReader(b *testing.B) {
 		})
 	}
 }
+
+func TestReaderReadSubset(t *testing.T) {
+	type Point3D struct { X, Y, Z int64 }
+	type Point2D struct { X, Y int64 }
+
+	f := func(points3D []Point3D) bool {
+		if len(points3D) == 0 {
+			return true
+		}
+		buf := new(bytes.Buffer)
+		err := writeParquetFile(buf, makeRows(points3D))
+		if err != nil {
+			t.Error(err)
+			return false
+		}
+		reader := parquet.NewReader(bytes.NewReader(buf.Bytes()))
```

Comment:
A `bytes.Buffer` implements `Read()` doesn't it? I don't think you need to call `.Bytes()` here

Reply:
Yes, but the signature is `parquet.NewReader(io.ReaderAt)` :)